### PR TITLE
736 limit image selection

### PR DESCRIPTION
**Pull request by kevinhowbrook**
[Ticket 736](https://projects.torchbox.com/projects/rca-website-rebuild/tickets/736)

- Limits the image chooser queryset to images the request.user > student_page > image_collection has access to.

### CODE REVIEW

File: rca/account_management/views.py [hunk 1]
```diff
@@ -107,13 +107,17 @@ def form_valid(self, form):
...
+                    live=True,
                 )
             )
             # Place the page in draft mode
-            StudentPage.objects.filter(student_user_account=student_user).update(
-                live=False
-            )
+            student_page = StudentPage.objects.get(student_user_account=student_user)
+            student_page.save_revision()
+            student_page.unpublish()
```
[the remark this answers is not in the document]
Comment (richbrennan) [reply rca/account_management/views.py:121]: That makes sense :+1:

**Review by richbrennan — COMMENTED** [review]
A few comments, thanks Kevin

File: rca/images/wagtail_hooks.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+from rca.people.models import StudentPage
+
+
+@hooks.register("construct_image_chooser_queryset")
+def show_my_uploaded_images_only(images, request):
+    # Only show uploaded images
+    user = request.user
+    # is this user a student?
+    if not user.groups.filter(name="Students").exists():
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/images/wagtail_hooks.py:11]: yea - historical code/rebase issues. I'll update that

File: rca/images/wagtail_hooks.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        return images
+
+    try:
+        # Look up their student page
+        student_page = StudentPage.objects.get(student_user_account=user)
+        # Get the image collection the user is linked to through their student page
+        student_collection = student_page.student_user_image_collection
+    except StudentPage.DoesNotExist:
+        # this is a student, but a page doesn't exists
+        return images
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/images/wagtail_hooks.py:21]: I'l change this to your suggestion, returning `images.none()`. If they don't have a page they can't really do anything on the admin side

**Pull request merged**